prog/download_boot_image.rb

```diff
@@ -63,9 +80,6 @@ def blob_storage_client
   end
 
   label def start
-    # YYY: we can remove this once we enforce it in the database layer.
-    fail "Version is required" if version.nil?
```

Comment:
Someone can still execute `vm_host.download_boot_image("image-name", version: nil)`. Let's maintain this additional check until we impose a constraint on the database.

Reply:
Even if it is passed as nil we'll still get the default version since it will be nil

Reply:
Are you sure? It think `fetch` only checks the existing of the key. If it exits with `nil` value, it will continue with `nil` value.

```
    @version ||= frame.fetch("version") { default_boot_image_version(image_name) }
```

Reply:
You're right, I mistested it apparently. I've added it back and updated the tests accordingly.